pkg/net/net.go

```diff
@@ -165,3 +188,76 @@ func GetNetworkInterface() (string, error) {
 
 	return "", fmt.Errorf("No network interfaces found")
 }
+
+func lessThan(left, right net.IP) bool {
+	var l, r big.Int
+	l.SetBytes(left)
+	r.SetBytes(right)
+	if l.Cmp(&r) == -1 {
+		return true
+	}
+	return false
+}
+
+func greaterThan(left, right net.IP) bool {
+	var l, r big.Int
+	l.SetBytes(left)
+	r.SetBytes(right)
+	if l.Cmp(&r) == 1 {
+		return true
+	}
+	return false
+}
+
+func RangesOverlap(range1start, range1end, range2start, range2end net.IP) bool {
+
+	if lessThan(range1start, range2start) && lessThan(range2start, range1end) {
+		return true
+	}
+	if lessThan(range2start, range1start) && lessThan(range1start, range2end) {
+		return true
+	}
+	if range1start.Equal(range2start) || range1end.Equal(range2end) {
+		return true
+	}
+	if range1end.Equal(range2start) || range1start.Equal(range2end) {
+		return true
+	}
+	return false
+}
+
+func IsRangeInCIDR(start, end net.IP, cidr *net.IPNet) bool {
+	if cidr.Contains(start) || cidr.Contains(end) {
+		return true
+	}
+	return false
+}
+
+func RangeContains(start, end, ip net.IP) bool {
+	if x := start.To4(); x != nil {
+		start = x
+	}
+	if x := end.To4(); x != nil {
+		end = x
+	}
+	if x := ip.To4(); x != nil {
+		ip = x
+	}
+
+	if len(ip) != len(start) {
+		return false
+	}
+	if len(ip) != len(end) {
+		return false
+	}
+
+	if ip.Equal(start) || ip.Equal(end) {
+		return true
+	}
+	if greaterThan(ip, start) && lessThan(ip, end) {
+		return true
+	}
+	return false
+}
+
+//TODO: Create an IPRange class.
```

Comment:
I can improve all the functions that take a start/end by creating an iprange class.
However, given the time constraints, I'm going to submit like this and when time is available, go back and create an iprange class.